File: UIViewPager/src/course/examples/UI/ViewPager/ImageAdapter.java
package course.examples.UI.ViewPager;

import android.app.Fragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.support.v13.app.FragmentStatePagerAdapter;

// Manages Fragments holding ImageViews
public class ImageAdapter extends FragmentStatePagerAdapter {

	// List of IDs corresponding to the images
	private Integer[] mImageIds = { R.drawable.sample_1, R.drawable.sample_2,
			R.drawable.sample_3, R.drawable.sample_4, R.drawable.sample_5,
			R.drawable.sample_6, R.drawable.sample_7 };

	public ImageAdapter(FragmentManager fm) {
		super(fm);
	}

	@Override
	public Fragment getItem(int i) {
		Fragment fragment = new ImageHolderFragment();
		Bundle args = new Bundle();
		args.putInt(ImageHolderFragment.RES_ID, mImageIds[i]);
		args.putString(ImageHolderFragment.POS, String.valueOf(i));
		fragment.setArguments(args);
		return fragment;
	}

	@Override
	public int getCount() {
		return mImageIds.length;
	}

}

File: ExamplesKotlin/UIViewPager/app/src/main/java/course/examples/ui/viewpager/ImageAdapter.kt
package course.examples.ui.viewpager

import android.os.Bundle
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager


// Manages Fragments holding ImageViews
internal class ImageAdapter(private val ids: IntArray, fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    override fun getItem(i: Int): Fragment {
        val fragment = ImageHolderFragment()
        val args = Bundle()
        args.putInt(ImageHolderFragment.KEY_RES_ID, ids[i])
        args.putString(ImageHolderFragment.KEY_POS, i.toString())
        fragment.arguments = args
        return fragment
    }

    override fun getCount(): Int {
        return ids.size
    }

}